jastrow.py

```diff
@@ -1,20 +1,44 @@
 import numpy as np
 from func3d import GaussianFunction
 
-
-def eedist(configs):
+def eedist_old(configs):
+     """returns a list of electron-electron distances within a collection """
+     ne=configs.shape[1]
+     d=np.zeros((configs.shape[0],int(ne*(ne-1)/2),3))
+     c=0
+     for i in range(ne):
+         for j in range(i+1,ne):
+             d[:,c,:]=configs[:,j,:]-configs[:,i,:]
+             c+=1
+     return d
+
+
+def eedist(configs, nup, ndown):
     """returns a list of electron-electron distances within a collection """
     ne=configs.shape[1]
-    d=np.zeros((configs.shape[0],int(ne*(ne-1)/2),3))
-    c=0
+    d1=np.zeros((configs.shape[0],int(nup*(nup-1)/2),3))      # up-up case
+    d2=np.zeros((configs.shape[0],int(nup*ndown),3))          # up-down case
+    d3=np.zeros((configs.shape[0],int(ndown*(ndown-1)/2),3))  # down-down case
+    c1=0
+    c2=0
+    c3=0
+
+    # First electrons are spin up by convenction
     for i in range(ne):
         for j in range(i+1,ne):
-            d[:,c,:]=configs[:,j,:]-configs[:,i,:]
-            c+=1
-    return d
-
-
-def eidist(configs, coords):
+            if((i<nup) and (j<nup)):
```

Comment:
Don't put if statements inside loops. If you must, instead loop over each population individually. Like for i in range(nup), for j in range(i+1,nup) and so on.

Reply:
I changed this to loop over individual populations instead.